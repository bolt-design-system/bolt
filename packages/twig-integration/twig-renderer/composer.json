{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.20.0",
  "license": "MIT",
  "repositories": [
    {
      "type": "path",
	    "url": "../twig-extensions-shared"
	  },
    {
      "type": "path",
      "url": "../twig-extensions-compat"
    }
  ],
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.42.5",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
