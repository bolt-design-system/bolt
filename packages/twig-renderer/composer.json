{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.3.0",
  "license": "MIT",
  "require": {
    "bolt-design-system/core-php": "^2.3.0",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.38.2",
    "webmozart/path-util": "^2.3",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
