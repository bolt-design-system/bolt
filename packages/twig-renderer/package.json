{
  "name": "@bolt/twig-renderer",
  "version": "2.7.0",
  "license": "MIT",
  "description": "Renders Twig templates using an on-the-fly PHP rendering service.",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "keywords": [
    "design system",
    "bolt design system",
    "twig",
    "php"
  ],
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.12.1",
    "@bolt/build-utils": "^2.7.0",
    "@bolt/drupal-twig-extensions": "^2.7.0",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "setup": "composer run setup"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://boltdesignsystem.com",
  "gitHead": "3d198feea5f73c63913a956a2a077e229ad8d462"
}
