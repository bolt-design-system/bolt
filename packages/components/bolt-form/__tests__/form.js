import { render, renderString } from '@bolt/twig-renderer';

describe('<bolt-form> Component', () => {
  const inputs = [
    {
      placeholder: 'Enter a Title',
      type: 'text',
    },
    {
      placeholder: 'Enter email address',
      type: 'email',
    },
    {
      placeholder: 'Enter Password',
      type: 'password',
    },
    {
      placeholder: 'Enter Search Terms',
      type: 'search',
    },
    {
      placeholder: 'Enter Pin Nmber',
      type: 'number',
    },
    {
      placeholder: 'Enter Phone Number',
      type: 'tel',
    },
  ];

  const options = [
    {
      type: 'option',
      value: '',
      label: '- Select an option -',
    },
    {
      type: 'option',
      value: 'option-a',
      label: 'Option A',
    },
    {
      type: 'option',
      value: 'option-b',
      label: 'Option B',
    },
  ];

  inputs.forEach(async inputChoice => {
    test(`input with inputAttribute ${JSON.stringify(
      inputChoice.type,
    )}`, async () => {
      const results = await renderString(
        `{% extends "@bolt-components-form/form-element.twig" %}

          {% set inputAttributes = ${JSON.stringify(inputChoice)} %}

          {% set label %}
            {% include("@bolt-components-form/form-label.twig") with {
              "title": 'Title',
              "displayType": 'floating'
            } only %}
          {% endset %}

          {% set children %}
            {# Default input type is text #}
            {% set inputAttributes = inputAttributes|merge({'type': inputAttributes.type|default('text')}) %}

          {% include("@bolt-components-form/form-input.twig") with {
            "attributes": inputAttributes,
            "hasErrors": errors is not empty
          } only %}
        {% endset %}`,
      );
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`disabled input`, async () => {
    const results = await renderString(
      `{% extends "@bolt-components-form/form-element.twig" %}

          {% set inputAttributes = ${JSON.stringify({
            placeholder: "Don't try to edit this",
            disabled: 'disabled',
          })} %}

          {% set label %}
            {% include("@bolt-components-form/form-label.twig") with {
              "title": 'Title',
              "displayType": 'floating'
            } only %}
          {% endset %}

          {% set children %}
            {# Default input type is text #}
            {% set inputAttributes = inputAttributes|merge({'type': inputAttributes.type|default('text')}) %}

          {% include("@bolt-components-form/form-input.twig") with {
            "attributes": inputAttributes,
            "hasErrors": errors is not empty
          } only %}
        {% endset %}`,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`radio button`, async () => {
    const results = await renderString(
      `{% extends "@bolt-components-form/form-element.twig" %}

        {% set label %}
          {% include("@bolt-components-form/form-label.twig") with {
            "title": "Single Radio",
            "displayType": "inline-radio",
            "attributes": {
              "for": "radio"
            }
          } %}
        {% endset %}
        
        {% set children %}
          {% include("@bolt-components-form/form-input.twig") with {
            "attributes": {
              "type": "radio",
              "id": "radio",
              "name": ""
            },
        } %}
      {% endset %}`,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`checkbox`, async () => {
    const results = await renderString(
      `{% extends "@bolt-components-form/form-element.twig" %}

        {% set label %}
          {% include("@bolt-components-form/form-label.twig") with {
            "title": "Single Checkbox",
            "displayType": "inline-checkbox",
            "attributes": {
              "for": "checkbox"
            }
          } %}
        {% endset %}
        
        {% set children %}
          {% include("@bolt-components-form/form-input.twig") with {
            "attributes": {
              "type": "checkbox",
              "id": "checkbox",
              "name": ""
            },
        } %}
      {% endset %}`,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`textarea`, async () => {
    const results = await renderString(
      `{% extends "@bolt-components-form/form-element.twig" %}
      
      {% set label %}
        {% include("@bolt-components-form/form-label.twig") with {
          "title": "Job Description",
          "displayType": "floating"
        } %}
      {% endset %}
      
      {% set children %}
        {% include("@bolt-components-form/form-textarea.twig") with {
          "attributes": {
            "placeholder": "Describe the job",
            "required": true
          }
        } %}
      {% endset %}`,
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test(`selection list`, async () => {
    const results = await render('@bolt-components-form/form-select.twig', {
      options,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
