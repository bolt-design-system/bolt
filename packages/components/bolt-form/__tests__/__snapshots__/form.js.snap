// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-form> Component checkbox 1`] = `
<div class="c-bolt-input-list__item">
  <input type="checkbox"
         id="checkbox"
         name
         class="c-bolt-input c-bolt-input--checkbox"
  >
  <label for="checkbox"
         class="c-bolt-inline-label c-bolt-inline-label--checkbox"
  >
    Single Checkbox
  </label>
</div>
`;

exports[`<bolt-form> Component disabled input 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Don&#039;t try to edit this"
         disabled="disabled"
         type="text"
         class="c-bolt-input c-bolt-input--text"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "email" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter email address"
         type="email"
         class="c-bolt-input c-bolt-input--email"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "number" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter Pin Nmber"
         type="number"
         class="c-bolt-input c-bolt-input--number"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "password" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter Password"
         type="password"
         class="c-bolt-input c-bolt-input--password"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "search" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter Search Terms"
         type="search"
         class="c-bolt-input c-bolt-input--search"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
        <li class="o-bolt-inline-list__item">
          <button class="c-bolt-input-icon c-bolt-input-icon--link"
                  type="submit"
          >
            <bolt-icon name="search"
                       size="medium"
            >
            </bolt-icon>
          </button>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "tel" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter Phone Number"
         type="tel"
         class="c-bolt-input c-bolt-input--tel"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component input with inputAttribute "text" 1`] = `
<div class="c-bolt-input-list__item">
  <input placeholder="Enter a Title"
         type="text"
         class="c-bolt-input c-bolt-input--text"
  >
  <div class="c-bolt-input-icons">
    <bolt-input-icons>
      <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
        <li class="o-bolt-inline-list__item">
          <div class="c-bolt-input-icon--invalid">
            <bolt-icon name="warning"
                       size="medium"
            >
            </bolt-icon>
          </div>
        </li>
      </ul>
    </bolt-input-icons>
  </div>
  <label class="c-bolt-floating-label">
    Title
  </label>
</div>
`;

exports[`<bolt-form> Component radio button 1`] = `
<div class="c-bolt-input-list__item">
  <input type="radio"
         id="radio"
         name
         class="c-bolt-input c-bolt-input--radio"
  >
  <label for="radio"
         class="c-bolt-inline-label c-bolt-inline-label--radio"
  >
    Single Radio
  </label>
</div>
`;

exports[`<bolt-form> Component selection list 1`] = `
<select class="c-bolt-input c-bolt-input--select">
  <option value>
    - Select an option -
  </option>
  <option value="option-a">
    Option A
  </option>
  <option value="option-b">
    Option B
  </option>
</select>
<div class="c-bolt-input-icons">
  <bolt-input-icons>
    <ul class="o-bolt-inline-list o-bolt-inline-list--xsmall">
      <li class="o-bolt-inline-list__item">
        <div class="c-bolt-input-icon--invalid">
          <bolt-icon name="warning"
                     size="medium"
          >
          </bolt-icon>
        </div>
      </li>
      <li class="o-bolt-inline-list__item">
        <div class="c-bolt-input-icon">
          <bolt-icon name="chevron-down"
                     size="medium"
          >
          </bolt-icon>
        </div>
      </li>
    </ul>
  </bolt-input-icons>
</div>
`;

exports[`<bolt-form> Component textarea 1`] = `
<div class="c-bolt-input-list__item">
  <textarea placeholder="Describe the job"
            required
            class="c-bolt-input"
  >
  </textarea>
  <label class="c-bolt-floating-label">
    Job Description
  </label>
</div>
`;
